crates/hir_ty/src/tests/simple.rs

```diff
@@ -28,6 +28,30 @@ mod boxed {
     );
 }
 
+#[test]
+fn infer_box_with_allocator() {
+    check_types(
+        r#"
+//- /main.rs crate:main deps:std
+fn test() {
+    let x = box 1;
+    let t = (x, box x, box &1, box [1]);
+    t;
+} //^ (Box<i32, {unknown}>, Box<Box<i32, {unknown}>, {unknown}>, Box<&i32, {unknown}>, Box<[i32; _], {unknown}>)
```

Comment:
I don't think this test is enough to trigger the crash? it might be necessary to add some trait method call to make it actually panic

Reply:
It is enough to make sure we accept a variable number of parameters for `Box` (otherwise, the `{unknown}` would not be there). I was able to reproduce the crash when forcing 2 arguments and running the `infer_box` test.

I tried to define a mock Global allocator, but it doesn't seem to be resolved and always appear as `unknown`.